migrations for adding new coloumns to systemConfigs

deepikasl opened the pull request:
https://github.com/Shippable/base/issues/848

This should be merged after: [api/pull/16609](https://github.com/Shippable/api/pull/16609)

**locally tested.**
![screenshot from 2017-01-09 11 12 32](https://cloud.githubusercontent.com/assets/18304961/21757982/99bef672-d65c-11e6-909a-2b616eb3bfed.png)


```
wwwPort: {
    type: Sequelize.INTEGER,
    allowNull: false,
    field: 'wwwPort'
  },
redisUrl: {
    type: Sequelize.STRING(255),
    allowNull: false,
    field: 'redisUrl'
  },
awsAccountId: {
    type: Sequelize.STRING(12),,
    field: 'awsAccountId'
  },
jobConsoleBatchSize: {
    type: Sequelize.INTEGER,
    allowNull: false,
    field: 'jobConsoleBatchSize'
  },
jobConsoleBufferTimeInterval: {
    type: Sequelize.INTEGER,
    allowNull: false,
    field: 'jobConsoleBufferTimeInterval'
  },
defaultCronLoopHours: {
    type: Sequelize.INTEGER,
    allowNull: false,
    field: 'defaultCronLoopHours'
  },
apiRetryInterval: {
    type: Sequelize.INTEGER,
    allowNull: false,
    field: 'apiRetryInterval'
  },
vortexUrl: {
    type: Sequelize.STRING(255),
    allowNull: false,
    field: 'vortexUrl'
  },
cloudProviders: {
    type: Sequelize.STRING(80),
    allowNull: false,
    field: 'cloudProviders'
  },
truck: {
    type: Sequelize.BOOLEAN,
    allowNull: false,
    field: 'truck'
  },
hubspotTimeLimit: {
    type: Sequelize.INTEGER,
    allowNull: false,
    field: 'hubspotTimeLimit'
  },
hubspotListId: {
    type: Sequelize.INTEGER,
    allowNull: false,
    field: 'hubspotListId'
  },
hubspotShouldSimulate: {
    type: Sequelize.BOOLEAN,
    allowNull: false,
    field: 'hubspotShouldSimulate'
  }
```

shrivara commented:
We should only change `usr/system_configs.sql.template` here. The first thing we do before booting up API is to run this file. If we want to set a field to NOT NULL, we need to alter the table once we have populated the fields at the end of the script.

shrivara commented on migrations/v5.1.1.sql:
```diff
@@ -2238,7 +2238,72 @@ do $$
     end if;
 
-    --adds column dynamicNodesSystemIntegrationId in systemConfigs table
+    -- Add wwwPort column to systemConfigs
```
`systemConfigs.sql` will always be run first, so this is redundant. We can remove the changes made to the migrations.

shrivara commented on usr/system_configs.sql.template:
```diff
@@ -77,6 +90,71 @@ do $$
 
+    if not exists (select 1 from information_schema.columns where table_name = 'systemConfigs' and column_name = 'wwwPort') then
+      alter table "systemConfigs" add column "wwwPort" INTEGER;
+      -- add default value
```
We're not adding a default value right?

shrivara commented on usr/system_configs.sql.template:
```diff
@@ -115,7 +193,20 @@ do $$
             "consoleCleanupHour"={{CONSOLE_CLEANUP_HOUR}},
-            "customHostDockerVersion"='{{CUSTOM_HOST_DOCKER_VERSION}}'
+            "customHostDockerVersion"='{{CUSTOM_HOST_DOCKER_VERSION}}',
+            "wwwPort"= {{WWW_PORT}},
```
Remove extra space. Let's keep this consistent.

shrivara commented:
@deepikasl We'll need to document these changes somewhere so, we're aware of this for the next release. Also, before merging this, we need to add these values to alpha, rc. Could you do that?

deepikasl commented:
Added following fields in rc state.json: 
```
   "wwwPort": 50001,
    "awsAccountId": null,
    "jobConsoleBatchSize": 10,
    "jobConsoleBufferTimeInterval": 3000,
    "defaultCronLoopHours": 2,
    "apiRetryInterval": 3,
    "vortexUrl": "'https://rcapi.shippable.com/vortex',
    "cloudProviders": 'ec2',
    "truck": true,
    "hubspotTimeLimit": 10,
    "hubspotListId": 100,
    "hubspotShouldSimulate": "true"
```

Added following fields in alpha state.json: 
```
   "wwwPort": 50001,
    "awsAccountId": null,
    "jobConsoleBatchSize": 10,
    "jobConsoleBufferTimeInterval": 3000,
    "defaultCronLoopHours": 2,
    "apiRetryInterval": 3,
    "vortexUrl": "'https://alphaapi.shippable.com/vortex',
    "cloudProviders": 'ec2',
    "truck": true,
    "hubspotTimeLimit": 10,
    "hubspotListId": 100,
    "hubspotShouldSimulate": "true"
```

shrivara commented:
Why did we change `hubspotShouldSimulate` to string?

shrivara commented on usr/system_configs.sql.template:
```diff
@@ -195,8 +286,74 @@ do $$
+      alter table "systemConfigs" alter column "truck" set not null;
+    end if;
+
+    if exists (select 1 from information_schema.columns where table_name = 'systemConfigs' and column_name = 'hubspotShouldSimulate') then
```
Hubspot fields should be nullable, not every install is going to use it.

shrivara replied:
`cloudProviders` also maybe?

deepikasl commented:
@shrivara: here for setting up hubspot, https://github.com/Shippable/micro/blob/master/nod/hubspotSync/_common/setupHubspotSync.js#9, we are comparing with string and also previously it is a string.

shrivara commented:
Let's update hubspotSync to handle it correctly then, this should be boolean.

deepikasl commented:
Made `hubspotShouldSimulate` BOOLEAN.
